python/ql/src/semmle/python/frameworks/Ujson.qll

```diff
@@ -0,0 +1,76 @@
+/**
+ * Provides classes modeling security-relevant aspects of the `ujson` PyPI package.
+ * See https://pypi.org/project/ujson/.
+ */
+
+private import python
+private import semmle.python.dataflow.new.DataFlow
+private import semmle.python.dataflow.new.TaintTracking
+private import semmle.python.Concepts
+private import semmle.python.ApiGraphs
+
+/**
+ * Provides models for the `ujson` PyPI package.
+ * See https://pypi.org/project/ujson/.
+ */
+private module UjsonModel {
+  /**
+   * A call to `usjon.dumps` or `ujson.encode`.
+   */
+  private class UjsonDumpsCall extends Encoding::Range, DataFlow::CallCfgNode {
+    UjsonDumpsCall() { this = API::moduleImport("ujson").getMember(["dumps", "encode"]).getACall() }
+
+    override DataFlow::Node getAnInput() { result in [this.getArg(0), this.getArgByName("obj")] }
+
+    override DataFlow::Node getOutput() { result = this }
+
+    override string getFormat() { result = "JSON" }
+  }
+
+  /**
+   * A call to `ujson.dump`.
+   */
+  private class UjsonDumpCall extends Encoding::Range, DataFlow::CallCfgNode {
+    UjsonDumpCall() { this = API::moduleImport("ujson").getMember("dump").getACall() }
+
+    override DataFlow::Node getAnInput() { result = this.getArg(0) }
+
+    override DataFlow::Node getOutput() {
+      result.(DataFlow::PostUpdateNode).getPreUpdateNode() = this.getArg(1)
```

Comment:
I feel like this is clunky enough that we should extend the API to make it easy to access the `PostUpdateNode` (if it it exists) for a given `Node`. 

That is, something like
```ql
PostUpdateNode getPostUpdateNode() { result.getPreUpdateNode() = this } 
```
on the `Node` class itself, so we could just write `result = this.getArg(1).getPostUpdateNode()` above.

Reply:
That would be nice indeed :+1:

Reply:
I did this in a PR: https://github.com/github/codeql/pull/5926